runtime/permissions.rs

```diff
@@ -204,7 +205,33 @@ impl fmt::Display for NetDescriptor {
 }
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
-pub struct EnvDescriptor(pub String);
+pub struct EnvDescriptor {
+  env: String,
+}
```

Comment:
Am I breaking some deserialization with this change?

Reply:
You can still make that work, by just removing `pub` modifier before the `String` type?

Reply:
I guess because this is mostly constructed in this file there wouldn't be anything preventing someone from still doing `EnvDescriptor(...)`. Writing `EnvDescriptor { env: ... }` is harder to do by accident.

Reply:
Sounds good

Reply:
The other Descriptors are all a tuple struct, even the ones that have a "constructor" function. please keep consistency. dont see much point for this change...

Reply:
@crowlKats it's to prevent accidentally writing `EnvDescriptor(...)`, which would be a bug. Maybe we could move this out to a separate file to prevent this module from doing that?

Reply:
> it's to prevent accidentally writing EnvDescriptor(...), which would be a potential bug if they didn't uppercase on windows.

You can keep it a tuple struct and just remove the `pub` if you really don't want people to use direct initialization of the struct. but even for that i see little point. Also if you think changing from a tuple struct to an "object" struct would make people less error prone, then you are mistaken. 
If people don't know the code they are working with, then that is on them. What we could and should do is document things, that would actually be a helpful solution to prevent misunderstandings and wrong usage of code.

Reply:
> You can keep it a tuple struct and just remove the pub if you really don't want people to use direct initialization of the struct.

See above. It would still be accessible within this module.

>  Also if you think changing from a tuple struct to an "object" struct would make people less error prone, then you are mistaken.

It is less error prone to accidentally write in this module. It stands out like a sore thumb to type or see `EnvDescriptor { env: ... }`. The difference between `EnvDescriptor(...)` and `EnvDescriptor::new(...)` is not as obvious.

I don't see the big deal here as we are talking about private API to this module (would be private API if we don't use `pub` on a tuple struct) and making it a object struct is not a foolproof, but a bug mitigation measure.

Honestly, I would rather change all these structs to be object structs so they describe their properties. Tuple structs aren't great. For example looking at `pub struct NetDescriptor(pub String, pub Option<u16>);`... you have to look at the implementation to find out that the second argument is a port.